Front-end/client/src/components/App/Widgets/Weather/Weather.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import LoadingSpinner from "../../../../utils/LoadingSpinner";
+import "./Weather.css";
+const Weather = props => {
+  /*
+    First We Fetch Our Location and set it to our state
+    Then we fetch the weather data
+    then we display the current weather
+
+   */
+  const [location, setLocation] = useState({
+    latitude: 0,
+    longitude: 0,
+    city: null,
+    state: null,
+  });
+  const [weatherData, setWeatherData] = useState({
+    main: null,
+    clouds: null,
+    temp: null,
+    wind: null,
+    rain: null,
+    description: null,
+  });
+  const [error, setError] = useState(null);
+
+  const fetchLocation = () => {
+    var options = {
+      enableHighAccuracy: true,
+      timeout: 5000,
+      maximumAge: 0,
+    };
+
+    function success(pos) {
+      var crd = pos.coords;
+      let latitude = crd.latitude;
+      let longitude = crd.longitude;
+
+      //Here we take the coordinates we get from the native web api and use them to get the city those coords lead to
+      let reverseGeocodeKey = "j5l5gEtlUQow2kzyIZmI0BvcAf4xo7By";
+      axios
+        .get(
+          `http://www.mapquestapi.com/geocoding/v1/reverse?key=${reverseGeocodeKey}&location=${latitude},${longitude}&includeRoadMetadata=true&includeNearestIntersection=true`,
```

Comment:
This piece of code takes our coords we get and feeds them to a reverse geocoding api to fetch the city location